main/src/ca/uwaterloo/flix/api/Flix.scala

```diff
@@ -540,7 +541,24 @@ class Flix {
         this.cachedResolverAst = afterResolver
         this.cachedTyperAst = afterTyper
       }
-      output
+      errors ++= readerErrors
```

Comment:
This only gets you the errors on the happy path, but if resolver fails you lose any errors that occurred earlier. 

I think you need to get rid of the for-comprehension and then add the errors to the error set as you go.

Reply:
This is not actually the issue, but I guess it is better to remove the for-comprehension now rather than later

Reply:
The issue was that it now returned `HardFailure` only containing errors when the parser recovery tests assumed that it would get a soft failure which contains a root

Reply:
Kind of confused. Does this not make the parser tests less valuable or?

Reply:
Perhaps there is no alternative?

Reply:
From the documentation of `TestParser`

> Note that these tests use check rather than compile.
> That's because a compile converts any check failure into a HardFailure before running, codegen so the result we would like to expect is lost.

Reply:
So they want to assert that the front-end succeeded but with errors whereas there are other tests that assert that `compile` fails. However, `compile` removes the `root` if the front-end fails